packages/amplify-graphql-api-construct/.jsii

```diff
@@ -4879,19 +4879,38 @@
         {
           "abstract": true,
           "docs": {
+            "deprecated": ", use dataStoreConfiguration instead.",
             "remarks": "For more information, refer to https://docs.amplify.aws/lib/datastore/getting-started/q/platform/js/",
-            "stability": "stable",
+            "stability": "deprecated",
             "summary": "Configure conflict resolution on the Api, which is required to enable DataStore Api functionality."
           },
           "immutable": true,
           "locationInModule": {
             "filename": "src/types.ts",
-            "line": 640
+            "line": 647
           },
           "name": "conflictResolution",
           "optional": true,
           "type": {
-            "fqn": "@aws-amplify/graphql-api-construct.ConflictResolution"
+            "fqn": "@aws-amplify/graphql-api-construct.DataStoreConfiguration"
```

Comment:
This worries me a little bit because JSII now declares that the `conflictResolution` field has a new type, as opposed to declaring the existing type or describing the existing shape.. I don't think it'll be a problem for TS, but when we start supporting more strictly typed languages we might need to keep an eye on this kind of change. 

In the meantime, can you confirm that you can build a standalone project using the packaged Data construct, using both `dataStoreConfiguration` and `conflictResolution` fields? I'd like this to be totally separate from the E2E tests, so we're not relying on any inside project configs. I don't think it's necessary to commit it, but I do want to validate that things work for existing customers without anything more dramatic than a deprecation message.

Reply:
Looks like it could be an issue: https://aws.github.io/jsii/user-guides/lib-author/typescript-restrictions/#type-aliases

I've thought of another option that would give equivalent types.

```typescript
export interface ConflictResolution extends DataStoreConfiguration {}
```

I'll test it out an report back.

Reply:
Works fine for all cases.

```typescript
export class BackendStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // also works if type is DataStoreConfiguration
    const conflictResolution: ConflictResolution = 
      {
        models: {
          Blog: {
            handlerType: 'AUTOMERGE',
            detectionType: 'VERSION',
          }
        }
      };

    const amplifyApi = new AmplifyGraphqlApi(this, "MyNewApi", {
      definition: AmplifyGraphqlDefinition.fromFiles(path.join(__dirname, "schema.graphql")),
      authorizationModes: {
        defaultAuthorizationMode: 'API_KEY',
        apiKeyConfig: {
          expires: cdk.Duration.days(30)
        }
      },
      
      // works with either ConflictResolution or DataStoreConfiguration type
      // works when dataStoreConfiguration keys is used with either type
      conflictResolution: conflictResolution,
    })
  }
}

```